trusted-types/support/csp-violations.js

```diff
@@ -33,10 +39,22 @@ function trusted_type_violations_and_exception_for(fn) {
 
     // Run the specified function and record any exception.
     try {
-      fn();
+      new EventSource("/common/blank.html");
     } catch(e) {
-      result.exception = e;
+      if (!e instanceof DOMException || e.name !== "SecurityError") {
+        throw e;
+      }
```

Comment:
Also here, I meant to factor out this (+ the comment for webkit) in a `forceConnectSrc()` function so it can be shared here and below.